File: app/src/main/java/de/xikolo/jobs/GetItemWithContentJob.java
package de.xikolo.jobs;

import android.util.Log;

import com.birbit.android.jobqueue.Params;

import de.xikolo.config.Config;
import de.xikolo.jobs.base.BaseJob;
import de.xikolo.jobs.base.JobCallback;
import de.xikolo.models.base.Sync;
import de.xikolo.managers.UserManager;
import de.xikolo.models.Item;
import de.xikolo.network.ApiService;
import de.xikolo.utils.NetworkUtil;
import retrofit2.Response;

public class GetItemWithContentJob extends BaseJob {

    public static final String TAG = GetItemWithContentJob.class.getSimpleName();

    private String itemId;

    public GetItemWithContentJob(JobCallback callback, String itemId) {
        super(new Params(PRIORITY_HIGH), callback);

        this.itemId = itemId;
    }

    @Override
    public void onAdded() {
        if (Config.DEBUG) {
            Log.i(TAG, TAG + " added | item.id " + itemId);
        }
    }

    @Override
    public void onRun() throws Throwable {
        if (NetworkUtil.isOnline()) {
            if (UserManager.isAuthorized()) {
                final Response<Item.JsonModel> response = ApiService.getInstance().getItemWithContent(
                        UserManager.getTokenAsHeader(),
                        itemId
                ).execute();

                if (response.isSuccessful()) {
                    if (Config.DEBUG)
                        Log.i(TAG, "Item received");

                    Sync.Data.with(Item.class, response.body())
                            .saveOnly()
                            .run();
                    syncItemContent(response.body());

                    if (callback != null) callback.success();
                } else {
                    if (Config.DEBUG) Log.e(TAG, "Error while fetching section list");
                    if (callback != null) callback.error(JobCallback.ErrorCode.ERROR);
                }
            } else {
                if (callback != null) callback.error(JobCallback.ErrorCode.NO_AUTH);
            }
        } else {
            if (callback != null) callback.error(JobCallback.ErrorCode.NO_NETWORK);
        }
    }

}

File: app/src/main/java/de/xikolo/network/jobs/GetItemWithContentJob.kt
package de.xikolo.network.jobs

import android.util.Log
import de.xikolo.config.Config
import de.xikolo.network.ApiService
import de.xikolo.network.jobs.base.ItemSyncHelper
import de.xikolo.network.jobs.base.NetworkJob
import de.xikolo.network.jobs.base.NetworkStateLiveData
import de.xikolo.network.sync.Sync
import ru.gildor.coroutines.retrofit.awaitResponse

class GetItemWithContentJob(private val itemId: String, networkState: NetworkStateLiveData, userRequest: Boolean) : NetworkJob(networkState, userRequest, Precondition.AUTH) {

    companion object {
        val TAG: String = GetItemWithContentJob::class.java.simpleName
    }

    override suspend fun onRun() {
        val response = ApiService.instance.getItemWithContent(itemId).awaitResponse()

        if (response.isSuccessful && response.body() != null) {
            if (Config.DEBUG) Log.i(TAG, "Item received")

            ItemSyncHelper.syncItemContent(response.body()!!)

            Sync.Data.with(response.body()!!)
                .saveOnly()
                .run()

            success()
        } else {
            if (Config.DEBUG) Log.e(TAG, "Error while fetching item")
            error()
        }
    }

}
